main.nf

```diff
@@ -1369,11 +1367,68 @@ if (!params.skip_compute && (protein_input || params.reference_proteome_fasta)){
       sourmash sig describe --csv ${csv} ${sig}
     """
     }
-  sourmash_sketches_peptide = sourmash_sketches_all_peptide.filter{ it[4].size() > 0 }
+  sourmash_sketches_peptide = sourmash_sketches_all_peptide.filter{ it[3].size() > 0 }
 } else {
   sourmash_sketches_peptide = Channel.empty()
 }
 
+if (params.bam || params.tenx_tgz) {
+  // Merge signatures from same sample id and sketch id
+
+  sourmash_sketches_nucleotide
+    .mix ( sourmash_sketches_peptide )
+    .set { ch_sourmash_sketches_mixed}
+
+  ch_fastq_id_to_cell_id_is_aligned
+    .combine ( ch_sourmash_sketches_mixed )
+    .dump( tag: 'fastq_id_to_cells__join__sketches' )
+    .groupTuple( by: 1 )
+    .dump( tag: 'fastq_id_to_cells__join__sketches__grouptuple' )
+    .set { ch_sourmash_sketches_to_merge }
+
+  process sourmash_sig_merge {
+    tag "${sig_id}"
+    label "low_memory"
+    publishDir "${params.outdir}/sketches_merged/${sketch_id}", mode: "${params.publish_dir_mode}",
+        saveAs: {filename ->
+            if (filename.indexOf(".csv") > 0) "description/$filename"
+            else if (filename.indexOf(".sig") > 0) "sigs/$filename"
+            else null
+        }
+
+    input:
+    set val(molecule), val(ksize), val(sketch_style), val(sketch_value), val(sample_id), file(reads) from ch_sourmash_sketches_to_merge
+
+    output:
+    file(csv) into ch_sourmash_sig_describe_merged
+    set val(sketch_id), val(molecule), val(ksize), val(sketch_value), file(sig) into sourmash_sketches
+
+    script:
+    // sketch_id = make_sketch_id(molecule, ksize, sketch_value, track_abundance, sketch_style)
+    sketch_value_flag = make_sketch_value_flag(sketch_style, sketch_value)
+    track_abundance_flag = track_abundance ? '--track-abundance' : ''
+    processes = "--processes ${task.cpus}"
+    sig_id = "${sample_id}__${sketch_id}"
+    sig = "${sig_id}.sig"
+    csv = "${sig_id}.csv"
+    """
+    sourmash compute \\
```

Comment:
so we are doing sourmash compute twice? is this also dependent on skip_compute flag?